src/core/lib/channel/channelz.cc

```diff
@@ -277,62 +280,64 @@ grpc_json* ServerNode::RenderJson() {
   return top_level_json;
 }
 
-SocketAddress::SocketAddress() : type_(AddressType::kUnset), port_(-1) {}
-
-SocketAddress::SocketAddress(SocketAddress&& other) {
-  type_ = other.type_;
-  port_ = other.port_;
-  blob_.reset((other.blob_.release()));
-}
-
-void SocketAddress::PopulateJson(const char* name, grpc_json* json) {
-  if (type_ == AddressType::kUnset) {
-    return;
-  }
+static void PopulateSocketAddressJson(grpc_json* json, const char* name,
+                                      const char* addr_str) {
+  if (addr_str == nullptr) return;
   grpc_json* json_iterator = nullptr;
   json_iterator = grpc_json_create_child(json_iterator, json, name, nullptr,
                                          GRPC_JSON_OBJECT, false);
   json = json_iterator;
   json_iterator = nullptr;
-  switch (type_) {
-    case AddressType::kTcpAddress:
-      json_iterator =
-          grpc_json_create_child(json_iterator, json, "tcpip_address", nullptr,
-                                 GRPC_JSON_OBJECT, false);
-      json = json_iterator;
-      json_iterator = nullptr;
-      json_iterator =
-          grpc_json_add_number_string_child(json, json_iterator, "port", port_);
-      json_iterator =
-          grpc_json_create_child(json_iterator, json, "ip_address", blob_.get(),
-                                 GRPC_JSON_STRING, false);
-      break;
-    case AddressType::kUdsAddress:
+  int port_num = -1;
+  grpc_uri* uri = grpc_uri_parse(addr_str, true);
+  if (uri != nullptr && (strcmp(uri->scheme, "fd") != 0)) {
+    const char* host_port = uri->path;
+    if (*host_port == '/') ++host_port;
```

Comment:
This line should only be executed for ipv4: and ipv6: schemes.  For the unix: scheme, we should not skip a leading slash, since it might be an absolute path.

Reply:
Done